Sources/SolToBoogie/ProcedureTranslator.cs

```diff
@@ -49,6 +78,7 @@ public ProcedureTranslator(TranslatorContext context, bool _genInlineAttrsInBpl
 
         public override bool Visit(ContractDefinition node)
         {
+            preTranslationAction(node);

```

Comment:
Seems that you are invoking preTranslationAction for each node type. Can this not be folded into one statement in visitor?

Reply:
Yeah, that's true. I tried to come up with a more elegant solution to this but it does not seem to be possible in the current infrastructure. In general, performing default actions inside a visitor is known to be problematic.

Reply:
Another solution would be to perform the gas instrumentation in a separate pass. But in this case, I would need to keep a mapping from every solidity AST node to its first Boogie statement. Is there an easy way to calculate this?